### ISSUE TITLE
Add storage: volatile: true

### ISSUE THREAD

**Issue by cgwalters** [coreos/ignition#935]
Let's support something like:

```
storage:
  volatile: true
```

This would have an effect like passing the `systemd.volatile` kernel command line argument; see also https://github.com/ostreedev/ostree/pull/856

Basically we have this logic today in our "Live" path; this would generalize it so it could easily be used for cloud instances too.

In particuar, I can see this being used by openshift-install for the bootstrap node.  Then we could pretty easily get rid of the bootstrap node requirement and have it re-Ignition itself into a control plane node.  This would *greatly* help the baremetal provisioning flow which does weird things with running bootstrap in a VM.

**Comment by cgwalters** [comment]
The implementation of this could just be writing out `/run/ostree-live`, and then we teach [fedora-coreos-config](https://github.com/coreos/fedora-coreos-config) how to handle that generically.  Or we could try to get agreement with systemd upstream to have `/run/systemd.volatile` or something that other Ignition users could also key off of.

**Comment by cgwalters** [comment]
@bgilbert raised in chat the issue of how this interacts with rebooting.  I think the basic answer here is - we wouldn't remove the `/boot/ignition.firstboot` flag, the same way we don't in the liveOS path.  So every boot would run Ignition.

**Comment by darkmuggle** [comment]
> The implementation of this could just be writing out `/run/ostree-live`, and then we teach [fedora-coreos-config](https://github.com/coreos/fedora-coreos-config) how to handle that generically. Or we could try to get an agreement with systemd upstream to have `/run/systemd.volatile` or something that other Ignition users could also key off of.

I'm not sure that we can use `systemd.volatile=yes`, since it only remounts `/sysroot`.  I think that the caused by `systemd-tmpfiles` but not sure.

``` 
bash-5.0# pwd
/sysroot/ostree/deploy/fedora-coreos/var/home
bash-5.0# find .
.
./core
./core/.bash_logout
./core/.bash_profile
./core/.bashrc
bash-5.0# find /home
/home
```

**Comment by cgwalters** [comment]
> I'm not sure that we can use systemd.volatile=yes

Agreed; a lot of the upstream systemd stuff is oriented towards mounting `/usr` as a filesystem, kind of glossing over how `/` is managed.  OSTree is much stricter with its `/etc` and `/var` are the only writable bits model.

**Comment by bgilbert** [comment]
This is a very OS-specific option, which mitigates against the feature, and it seems like this functionality could be implemented with kargs support (#1051) and/or writing a normal file.  Closing for now.

**Comment by cgwalters** [comment]
Moved this to https://github.com/coreos/fedora-coreos-tracker/issues/674
